From 6b63caf7d6bdded413985906276d023cfb623905 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 9 Jun 2015 10:32:28 +0200
Subject: [PATCH] util:bind_remount_recursive(): handle return 0 of
 set_consume()

set_consume() does not return -EEXIST, but 0, in case the key is already
in the Set.

(cherry picked from commit 85d834ae8e7d9e2c28ef8c1388e2913ed8fd0e3b)
---
 src/shared/util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/util.c b/src/shared/util.c
index 57782ba687..26eec38ea6 100644
--- a/src/shared/util.c
+++ b/src/shared/util.c
@@ -5135,7 +5135,7 @@ int bind_remount_recursive(const char *prefix, bool ro) {
                 while ((x = set_steal_first(todo))) {
 
                         r = set_consume(done, x);
-                        if (r == -EEXIST)
+                        if (r == -EEXIST || r == 0)
                                 continue;
                         if (r < 0)
                                 return r;
